source/common/http/filter/ip_tagging_filter.cc

```diff
@@ -9,7 +14,34 @@ IpTaggingFilter::~IpTaggingFilter() {}
 
 void IpTaggingFilter::onDestroy() {}
 
-FilterHeadersStatus IpTaggingFilter::decodeHeaders(HeaderMap&, bool) {
+FilterHeadersStatus IpTaggingFilter::decodeHeaders(HeaderMap& headers, bool) {
+  bool is_internal_request =
+      headers.EnvoyInternalRequest() && (headers.EnvoyInternalRequest()->value() ==
+                                         Headers::get().EnvoyInternalRequestValues.True.c_str());
+
+  if ((is_internal_request && config_->requestType() == FilterRequestType::EXTERNAL) ||
+      (!is_internal_request && config_->requestType() == FilterRequestType::INTERNAL) ||
+      (!config_->runtime().snapshot().featureEnabled("ip_tagging.http_filter_enabled", 100))) {
```

Comment:
Question: Why is `ip_tagger.filter_enabled` a runtime attribute? Seems this is a filter chain configuration decision.

Reply:
To be able to slowly roll out the feature, or turn it off in any emergency. It similar to the runtime key in the rate limit filter.

Reply:
Fair enough. I wonder if we should just have a general ability to skip filters using runtime or only enable them in x% of requests. That would avoid us having to bake into all the filters that need this. This is non-actionable, just thinking out loud :)

Reply:
I think that would be useful. Continuing the out loud thinking - Every service could have a default runtime baked into the Stream*Filter and the key would be based off of the service name.